src/test/java/io/personium/test/jersey/box/odatacol/UserDataCreateTest.java

```diff
@@ -759,6 +763,40 @@ public UserDataCreateTest() {
         }
     }
 
+    /**
+     * When_URL_isGivenAs__id_Then_ItShouldBe_Escaped.
+     */
+    @SuppressWarnings("unchecked")
+    @Test
+    public final void When_URL_isGivenAs__id_Then_ItShouldBe_Escaped() {
+        String userDataId = "https://server.exapmle/path?abc=def#ghi";
+        // Prepare Request Body
+        JSONObject body = new JSONObject();
+        body.put("__id", userDataId);
+        body.put("testword", "xxx");
+
+        try {
+            // make an request
+            TResponse response = createUserData(body, HttpStatus.SC_CREATED);
+
+            // get a result
+            JSONObject result = (JSONObject) ((JSONObject) response.bodyAsJson().get("d")).get("results");
+            userDataId = result.get("__id")
+                    .toString();
+            String uri = (String)((JSONObject)result.get("__metadata")).get("uri");
+
+            // key in the __metadata.uri should be encoded
+            //    not found in plain format
+            assertEquals(-1, uri.indexOf(userDataId));
+            //    found in encoded format
+            assertNotEquals(-1, uri.indexOf(CommonUtils.encodeUrlComp(userDataId)));
```

Comment:
It is better to assert with the expected uri like `https://cell.example.org/__/odata/Foo('https%3A%2F%2Faaa.example.org%2F')`.

Reply:
improved the test so that it can demonstrate the spec more explicitly.